### PR TITLE
insert java checks

### PR DESCRIPTION
**Pull request by dshimo**
#154 

- locate JRE and use it (Still looks for jars placed into `jars`)
- if java version is < 17, point to IBM Semeru download link

devCommand.ts changes were strictly fixing indent formatting

### CODE REVIEW

File: src/extension.ts [hunk 1]
```diff
@@ -163,83 +165,54 @@ export function registerFileWatcher(projectProvider: ProjectProvider): void {
...
+                    workspace.createFileSystemWatcher("**/bootstrap.properties"),
+                    workspace.createFileSystemWatcher("**/server.env")
+                ],
+            }
+        };
+
+        const serverOptions = prepareExecutable(LIBERTY_LS_JAR, requirements)
+
+        console.log("Creating new language client for Liberty Language Server");
+        libertyClient = new LanguageClient(LIBERTY_CLIENT_ID, "Language Support for Liberty", serverOptions, clientOptions);
```
Comment (kathrynkodama) [inline src/extension.ts:187]: I believe this is the text that appears in the "Output" tab correct? If so we should localize this message

File: src/extension.ts [hunk 2]
```diff
@@ -163,83 +165,54 @@ export function registerFileWatcher(projectProvider: ProjectProvider): void {
...
+                configurationSection: ["java", "[java]"],
+                fileEvents: [
+                    workspace.createFileSystemWatcher("**/*.java")
+                ],
+            }
+        };
+        const serverOptions = prepareExecutable(JAKARTA_LS_JAR, requirements);
+
+        console.log("Creating new language client for Jakarta Language Server")
+        jakartaClient = new LanguageClient(JAKARTA_CLIENT_ID, "Language Support for Jakarta", serverOptions, clientOptions);
```
Comment (kathrynkodama) [inline src/extension.ts:211]: Localization, should be "Language Support for Jakarta EE"

File: src/extension.ts [hunk 3]
```diff
@@ -163,83 +165,54 @@ export function registerFileWatcher(projectProvider: ProjectProvider): void {
...
+            synchronize: {
+                configurationSection: ["java", "[java]"],
+                fileEvents: [
+                    workspace.createFileSystemWatcher("**/*.java")
+                ],
+            }
+        };
+        const serverOptions = prepareExecutable(JAKARTA_LS_JAR, requirements);
+
+        console.log("Creating new language client for Jakarta Language Server")
```
Comment (kathrynkodama) [inline src/extension.ts:210]: Minor thing, but we should refer to as Language Server for Jakarta EE

File: src/util/requirements.ts [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+// https://github.com/redhat-developer/vscode-microprofile/blob/master/src/languageServer/requirements.ts
+
+/**
+ * Resolves the requirements needed to run the extension.
+ * Returns a promise that will resolve to a RequirementsData if
+ * all requirements are resolved, it will reject with ErrorData if
+ * if any of the requirements fails to resolve.
+ *
+ */
+export async function resolveRequirements(api: JavaExtensionAPI): Promise<RequirementsData> {
```
Comment (kathrynkodama) [inline src/util/requirements.ts:48]: There are few messages that need localization here

**Review by TrevCraw — APPROVED** [review]

**Review by kathrynkodama — APPROVED** [review]

File: src/extension.ts [hunk 5]
```diff
@@ -166,83 +168,54 @@ export function registerFileWatcher(projectProvider: ProjectProvider): void {
...
+        //Start up Liberty Language Server
+
+        // Options to control the language client
+        const clientOptions: LanguageClientOptions = {
+            documentSelector: SUPPORTED_LANGUAGE_IDS,
+            synchronize: {
+                configurationSection: ["properties", "plaintext"],
+                fileEvents: [
+                    workspace.createFileSystemWatcher("**/bootstrap.properties"),
+                    workspace.createFileSystemWatcher("**/server.env")
```
Comment (travis1111) [inline src/extension.ts:182]: should we only watch for 'target' and 'build' folders?

File: src/extension.ts [hunk 6]
```diff
@@ -166,83 +168,54 @@ export function registerFileWatcher(projectProvider: ProjectProvider): void {
...
-    return jakartaClient.onReady();
+function startupJakartaLangServer(context: ExtensionContext, api: JavaExtensionAPI) {
+    return resolveRequirements(api).then(requirements => {    
+        // Options to control the language client
+        const clientOptions: LanguageClientOptions = {
+            documentSelector: [{ scheme: "file", language: "java" }],
+            synchronize: {
+                configurationSection: ["java", "[java]"],
+                fileEvents: [
+                    workspace.createFileSystemWatcher("**/*.java")
```
Comment (travis1111) [inline src/extension.ts:207]: should we only watch for 'src'  folder?

**Review by travis1111 — CHANGES_REQUESTED** [review]

File: src/util/javaServerStarter.ts [hunk 7]
```diff
@@ -0,0 +1,71 @@
+/**
+ * Copyright 2019 Red Hat, Inc. and others.
```
[the remark this answers is not in the document]
Comment (kathrynkodama) [reply src/util/javaServerStarter.ts:2]: This code is re-used from https://github.com/redhat-developer/vscode-microprofile, so we have to keep the header of the original code.

[the remark this answers is not in the document]
Comment (travis1111) [reply src/util/javaServerStarter.ts:2]: Ah ic thanks

File: src/util/requirements.ts [hunk 8]
```diff
@@ -0,0 +1,152 @@
+/**
+ * Copyright 2019 Red Hat, Inc. and others.
```
[the remark this answers is not in the document]
Comment (travis1111) [reply src/util/requirements.ts:2]: @dshimo if requirements.ts is also reused from redhat, then please ignore my request.  Thanks!

File: src/extension.ts (hunk 6 above)
Comment (dshimo) [reply src/extension.ts:207]: Personally, it makes sense to me to not restrict the lsp4jakarta server to a specific folder in a workspace

**Review by travis1111 — APPROVED** [review]

**Pull request merged**